### PR TITLE
Random generators based on the STL random.

### PR DESCRIPTION
**Pull request by mdabek**
I've also added new exponential random generator, which is compatible with the exponential distribution. It generates random numbers from range [0; +inf) and takes single parameter, lambda -the  rate parameter.
The unit test bases on the distribution property, that the: E[X] = 1/lambda.
This PR is linked to the #14

### CODE REVIEW

File: include/fields2cover/utils/random.h [hunk 1]
```diff
@@ -52,6 +53,13 @@ class Random {
   /// @return Double \f$\in [min, max]\f$
   double getRandomExp(double min, double max);
 
+  /// Get a random double in a range [0;+inf) with exponential
+  /// probability.
+  /// Lower values are more probable than higher ones.
+  /// @param lambda is the rate parameter
+  /// @return Double \f$\in [min, max]\f$
```
Comment (Gonzalo-Mier) [inline include/fields2cover/utils/random.h:60]: This comment is not true

File: include/fields2cover/utils/random.h [hunk 2]
```diff
@@ -52,6 +53,13 @@ class Random {
...
+  double getRandomExp(double lambda);
```
Comment (Gonzalo-Mier) [inline include/fields2cover/utils/random.h:61]: getRandomExp do different things when has only one parameter than when it has two. One could expect the single parameter version to return a value between (0, lambda) which is not the case. Could you put another name to this function? Like getRandomExpDist(double lambda)

**Review by Gonzalo-Mier — CHANGES_REQUESTED** [review]

**Comment by Gonzalo-Mier** [comment]
Thanks for your contribution!!! Could you please add the same test you add it, but to the python tests? So both versions benefits from this new function 😁 Really good job

File: include/fields2cover/utils/random.h (hunk 1 above)
Comment (Gonzalo-Mier) [reply include/fields2cover/utils/random.h:60]: The @return part

File: include/fields2cover/utils/random.h (hunk 2 above)
Comment (mdabek) [reply include/fields2cover/utils/random.h:61]: Yes. I will change the method name.

File: include/fields2cover/utils/random.h (hunk 1 above)
Comment (mdabek) [reply include/fields2cover/utils/random.h:60]: Right. Will fix this.

**Review by Gonzalo-Mier — APPROVED** [review]

**Comment by Gonzalo-Mier** [comment]
Thank you again!

**Pull request merged**